deploy_tools/compile.py

```diff
@@ -0,0 +1,151 @@
+import json
+import os
+from typing import List, Dict
+
+from solc import compile_standard
+from eth_utils import add_0x_prefix
+
+from .files import find_files
+
+DEFAULT_OUTPUT_SELECTION = [
+    'abi',
+    'devdoc',
+    'userdoc',
+    'metadata',
+    'evm.bytecode'
+]
+
+
+def load_sources(file_paths: List[str]):
+    result = {}
+    for file_path in file_paths:
+        with open(file_path) as source_file:
+            result[file_path] = {
+                'content': source_file.read()
+            }
+    return result
+
+
+def normalize_contract_data(contract_data: Dict):
+    result = {}
+
+    for key, value in contract_data.items():
+        if key == 'evm':
+            evm_data = value
+
+            if 'bytecode' in evm_data:
+                result['bytecode'] = add_0x_prefix(evm_data['bytecode'].get('object', ''))
+
+            if 'deployedBytecode' in evm_data:
+                result['deployedBytecode'] = add_0x_prefix(evm_data['deployedBytecode'].get('object', ''))
+
+        elif key == 'metadata':
+            if value:
+                result[key] = json.loads(value)
+
+        else:
+            result[key] = value
+
+    return result
+
+
+def normalize_compiled_contracts(compiled_contracts: Dict, file_paths: List[str]):
+    result: Dict[str, Dict] = {}
+
+    for source_path, file_contracts in compiled_contracts.items():
+        if source_path not in file_paths:
+            continue
+
+        for contract_name, raw_contract_data in file_contracts.items():
+            contract_data = normalize_contract_data(raw_contract_data)
+            if contract_name not in result:
+                result[contract_name] = contract_data
+            else:
+                raise BaseException('Can not compile two contracts with the same name')
+
+    return result
+
+
+def log_compilation_errors(errors: List[Dict]):
+    for error in errors:
+        if 'formattedMessage' in error:
+            print(error['formattedMessage'])
+        else:
+            print(error['message'])
+
+
+def compile_project(
+        contracts_path: str = None,
+        *,
+        file_paths: List[str] = None,
+        allow_paths: List[str] = None,
+        pattern='*.sol'
+):
+    """
+    Compiles all contracts of the project into a single output
+    Args:
+        contracts_path: The path of the folder that includes the contracts
+        file_paths: Optional a list of to compiled contracts can be provided
```

Comment:
I would suggest changing this to `(optional) A list of contracts to be compiled`